src/components/Login/LoginContainer.tsx

```diff
@@ -91,11 +99,61 @@ export const InitialisationContainer: FunctionComponent<NavigationProps> = ({
     setAllProducts
   } = useProductContext();
   const { logout } = useLogout();
+  const lastResendWarningMessageRef = useRef("");
+  const { showAlert } = useContext(AlertModalContext);
 
   const resetStage = (): void => {
     setLoginStage("SCAN");
   };
 
+  const getResendConfirmationIfNeeded = async (): Promise<boolean> => {
+    return new Promise(resolve => {
+      if (!lastResendWarningMessageRef.current) {
+        resolve(true);
+      } else {
+        showAlert({
+          ...defaultConfirmationProps,
+          title: "Resend OTP?",
+          description: lastResendWarningMessageRef.current,
+          buttonTexts: {
+            primaryActionText: "Resend",
+            secondaryActionText: "No"
+          },
+          onOk: () => resolve(true),
+          onCancel: () => resolve(false),
+          visible: true
+        });
+      }
+    });
+  };
+
+  const handleRequestOTP = async (
+    fullNumber = mobileNumber
+  ): Promise<boolean> => {
+    try {
+      if (!(await getResendConfirmationIfNeeded())) return false;
+      const response = await requestOTP(fullNumber, codeKey, endpointTemp);
+      lastResendWarningMessageRef.current = response.warning ?? "";
+      return true;
+    } catch (e) {
+      if (e instanceof LoginLockedError) {
+        showAlert({
+          ...disabledAccessAlertProps,
+          description: e.message,
+          onOk: () => resetStage()
+        });
+      } else if (e instanceof LoginError) {
+        showAlert({
+          ...invalidEntryAlertProps,
+          description: e.message
+        });
+      } else {
+        alert(e);
```

Comment:
If it's an unknown error, do you think it'll be useful to show the ErrorBoundary or just show a generic warning alert?

Reply:
yeah, i think ErrorBoundary is more useful. will do that in another pr!